src/coreComponents/dataRepository/xmlWrapper.hpp

```diff
@@ -350,9 +350,10 @@ template< typename T >
 std::enable_if_t< traits::CanStreamInto< std::istringstream, T > >
 stringToInputVariable( T & target, string const & value, Regex const & regex )
 {
-  validateString( value, regex );
+  string_view stringTrimed = stringutilities::trimSpaces( value );
+  validateString( string( stringTrimed ), regex );
```

Comment:
I think you're getting issues in your unit tests because of the ordering of your validations:
```suggestion
  validateString( value, regex );
  string_view stringTrimed = stringutilities::trimSpaces( value );
```